content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+where `ct-id = Hash(Hash(session-id))`.
+
+
+# Session states
+
+A Noise session corresponding to a certain `session-id`:
+- is always **active** as long as it is not marked as **stale**. 
+For an active `session-id`, new messages are published on the content topic `/wakunoise/1/<ct-id>/proto`;  
+- is marked as **stale** if a [session termination message](https://rfc.vac.dev/spec/35/#session-termination-message) containing `Hash(session-id)` is published on the content topic `/wakunoise/1/<ct-id>/proto`. 
+Session information relative to stale sessions  MAY be deleted from users' device, unless required for later channel binding purposes.
+
+When a Noise session is marked as stale, it means that one party requested its termination while being online, 
+since publication of a hash pre-image for `ct-id` is required (i.e. `Hash(session-id)`).
+
+Currently, it is not possible to mark a Noise session as stale when `session-id` is lost or gets corrupted in users' devices.
+However, since `session-id` is shared between Alice and Bob, 
+one party MAY decide to mark a Noise session as stale if no message from the other end was received within a certain fixed time window. 
+
+The above mechanism allows a Noise session to be marked as stale either privately or publicly, 
+depending if `Hash(session-id)` is sent on `/wakunoise/1/<ct-id>/proto` to the other party in encrypted form or not, respectively. 
+
+When a Noise session is publicly marked as stale, 
+network peers MAY discard all [stored](https://rfc.vac.dev/spec/13/) messages addressed to the content topic `/wakunoise/1/<ct-id>/proto`. 
+In this the case and in order for parties to retrieve any eventually delayed message, 
+peers SHOULD wait a fixed amount of time before discarding stored messages corresponding to a stale Noise session. 
+
+A stale Noise session cannot be directly marked as active
+and parties are required to instantiate a new Noise session if they wish to communicate again. 
+
+However, parties can optionally include the `session-id` corresponding to a stale Noise session in the [prologue information](https://noiseprotocol.org/noise.html#prologue) employed in the Noise handshake they execute to instantiate their new Noise session. 
+This effectively emulates a mechanism to _"re-activate"_ a stale Noise session by binding it to a newly created active Noise session.
+
+In order to reduce users' metadata leakage, it is desirable (as suggested in [35/WAKU2-NOISE](https://rfc.vac.dev/spec/35/#after-handshake)) that content topics used for communications change every time a new message is exchanged.
+This can be easily realized by employing a key derivation function to compute a new `session-id` from the previously employed one (e.g. `session-id = HKDF(prev-session-id)`),
+while keeping the Inbound/outbound Cipher States, the content topic derivation mechanism and the stale mechanism the same as above.
+In this case, when one party sends **and** receives at least one message, 
```

Comment:
```suggestion
In this case, when one party sends **or** receives at least one message, 
```

Reply:
No, the **and** is correct. 

If, for example, a user sends a message and marks as stale all previous sessions, after a period of time store nodes may discard all messages corresponding to these stale sessions. If one party was temporary offline while the other party was sending many messages, with the exception of the last message, the others may not be delivered due to marking previous sessions stale. Once online again, he will then not be able to receive/decrypt messages due to these missed messages.

If, instead, a message is received after a message is sent, or vice-versa, it means that at that point the two party have synced session information, which allows them to encrypt/decrypt and move to new content topics. Hence, all session relative to messages exchanged before these last two can be marked as stale, since the session information is proved to be synced at least up to this point.